### PR TITLE
[^459] feat(hx-radio-control): surf-584 phase I radio form control

### PR DESCRIPTION
**Pull request by 100stacks**
* `Radio` (`<hx-radio-control>`) Phase I 

```html
<hx-radio-control>
  <input type="radio" id="rad1" />
  <label for="rad1">
    <hx-radio></hx-radio>
    Radio Button 1
  </label>
</hx-radio-control>
```

### LGTM's
- [x] Dev LGTM
- [x] Design LGTM
- [x] Zoom LGTM

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* processing.

Building with commit 2262bba1c2a25349cbc8c333058721da71e8e83e

https://app.netlify.com/sites/helix-ui/deploys/5c65ec1c5abc2100088ac89d

File: docs/components/radios/index.html [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+{% extends 'component.njk' %}
+{% block content %}
+  <section>
+    <h2 id="basic-radio">Basic Radio</h2>
+    <div class="example" id="vue-radioDemo" v-cloak>
+      <header>
+        <form class="beta-hxForm">
+          <fieldset>
+            <legend>Options</legend>
+            <label>
```
Comment (CITguy) [inline docs/components/radios/index.html]: Don't use inputs nested in labels. Try the following, instead:

```html
<div>
  <input id="chkIsChecked" v-model="isChecked" />
  <label for="chkIsChecked">Checked</label>
</div>
```

File: docs/components/radios/index.html [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            </label>
+            <label>
+              <input type="radio" name="radioDemo" v-model="isInvalid" />
+              Invalid
+            </label>
+          </fieldset>
+        </form>
+      </header>
+
+      <div class="hxRadio">
```
Comment (100stacks) [inline docs/components/radios/index.html]: wrap in another `<div>`

File: docs/components/radios/index.html [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          <input type="radio"
+            :checked="isChecked"
+            :disabled="isDisabled"
+            :invalid="isInvalid"
+            @change="onChange"
```
Comment (100stacks) [inline docs/components/radios/index.html]: remove `@change`

File: docs/components/radios/index.html [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            </label>
+          </fieldset>
+        </form>
+      </header>
+
+      <div class="hxRadio">
+          <input type="radio"
+            :checked="isChecked"
+            :disabled="isDisabled"
+            :invalid="isInvalid"
```
Comment (100stacks) [inline docs/components/radios/index.html]: remove `:invalid`

File: docs/components/radios/index.html [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+              <input type="radio" name="radioDemo" v-model="isInvalid" />
+              Invalid
+            </label>
+          </fieldset>
+        </form>
+      </header>
+
+      <div class="hxRadio">
+          <input type="radio"
+            :checked="isChecked"
```
Comment (100stacks) [inline docs/components/radios/index.html]: remove `:checked`

File: src/helix-ui/elements/HXRadioElement.html [hunk 6]
```diff
@@ -0,0 +1 @@
+<div id="hxRadio"></div>
```
Comment (100stacks) [inline src/helix-ui/elements/HXRadioElement.html]: remove file

File: src/helix-ui/elements/HXRadioElement.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import { HXElement } from './HXElement';
+import shadowMarkup from './HXRadioElement.html';
```
Comment (100stacks) [inline src/helix-ui/elements/HXRadioElement.js]: remove line 2 and 3

File: src/helix-ui/elements/HXRadioElement.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+ * @hideconstructor
+ * @since 0.16.0
+ */
+export class HXRadioElement extends HXElement {
+    static get is () {
+        return 'hx-radio';
+    }
+
+    static get template ()  {
+        return `<style>${shadowStyles}</style>${shadowMarkup}`;
```
Comment (100stacks) [inline src/helix-ui/elements/HXRadioElement.js]: return an empty string

File: src/helix-ui/elements/HXRadioElement.less [hunk 9]
```diff
@@ -0,0 +1,13 @@
+@import "./HXElement";
```
Comment (100stacks) [inline src/helix-ui/elements/HXRadioElement.less]: remove file

File: src/helix-ui/styles/components/radios.less [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+  > input[type="radio"] {
+    // visually hide
+    height: 1px;
+    width: 1px;
+    overflow: hidden;
+    // margin: -1px;
+
+    // Focus
+    &.pseudo-focus,
```
Comment (100stacks) [inline src/helix-ui/styles/components/radios.less]: remove the `.pseudo-*` classes...these will be in the `docs` directory

File: src/helix-ui/styles/components/radios.less [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+  // box-shadow: inset 0 0 0 1px black;
+  border: 1px dotted @cyan-500;
+
+  > label {
+    &:hover {
+      cursor: pointer;
+    }
+  }
+
+  > input[type="radio"] {
```
Comment (100stacks) [inline src/helix-ui/styles/components/radios.less]: style the before (everything else) and after (the tick)

File: docs/components/radios/index.html (hunk 1 above)
Comment (CITguy) [reply docs/components/radios/index.html]: Please revert changes to this file.

File: docs/docs.js [hunk 12]
```diff
@@ -19,6 +19,7 @@ import './components/modals/modal-demo';
 import './components/panels/panel-demo';
 import './components/pills/pill-demo';
 import './components/popovers/popover-demo';
+import './components/radios/radio-demo';
```
Comment (CITguy) [inline docs/docs.js]: Please revert changes, I'll handle this when I update docs.

File: docs/components/radios/radio-demo.js [hunk 13]
```diff
@@ -0,0 +1,40 @@
+import Util from '../../_util';
```
Comment (CITguy) [inline docs/components/radios/radio-demo.js]: Please remove this file. I'll handle this with updates to docs.

File: docs/components/radios/index.html [hunk 14]
```diff
@@ -9,6 +9,67 @@
 {% block content %}
   <section>
     <h2 id="basic-radio">Basic Radio</h2>
-    <p>TBD</p>
+    <div class="example" id="vue-radioDemo" v-cloak>
```
Comment (CITguy) [inline docs/components/radios/index.html]: Please revert changes to this file. I'll handle this when I update docs.

File: docs/elements/hx-radio/index.html [hunk 15]
```diff
@@ -6,5 +6,40 @@
 ---
 {% extends 'element.njk' %}
 {% block content %}
-  <p>TBD...</p>
+  <section>
```
Comment (CITguy) [inline docs/elements/hx-radio/index.html]: Please revert changes to this file. I'll handle this when I update docs.

File: src/helix-ui/styles/elements/hx-radio.less [hunk 16]
```diff
@@ -0,0 +1,10 @@
+hx-radio {
```
Comment (CITguy) [inline src/helix-ui/styles/elements/hx-radio.less]: Please remove file. Keep all styles in `components/radios.less` for this PR.

File: src/helix-ui/styles/elements.less [hunk 17]
```diff
@@ -39,6 +39,7 @@
 @import 'elements/hx-popover-foot';
 @import 'elements/hx-popover-head';
 @import 'elements/hx-progress';
+// @import 'elements/hx-radio';
```
Comment (CITguy) [inline src/helix-ui/styles/elements.less]: Please revert. Keep all styles in `components/radios.less` for this PR.

File: docs/components/radios/test.html [hunk 18]
```diff
@@ -11,6 +12,958 @@
...
+          </label>
+      </hx-radio-control>
+    </div>
+  </section>
+
+  <hr />
+
+  <section>
+    <header>
+      <h1>Radio State Table</h1>
```
Comment (cathysiller) [inline docs/components/radios/test.html]: remove `Radio State Table` heading

File: docs/components/radios/test.html [hunk 19]
```diff
@@ -11,6 +12,958 @@
...
+          <td>(none)</td>
+          <td>
+            <hx-radio-control>
+              <input
+                type="radio"
+                id="radOptionalEnabledUncheckedNoneValid"
+              />
+              <label for="radOptionalEnabledUncheckedNoneValid">
+                <hx-radio></hx-radio>
+                :optional:enabled:unchecked:none:valid
```
Comment (cathysiller) [inline docs/components/radios/test.html]: add focus styles in the tables on this page where we talked about on zoom!!

File: docs/styles/demo/radios-docs.less [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+        }
+      }
+
+      &:invalid.mock-focus {
+        &:extend(hx-radio-control > input[type="radio"]:invalid:focus all);
+      }
+
+      &:invalid.mock-hover {
+        &:extend(hx-radio-control > input[type="radio"]:invalid:hover all);
+        // color: green;
```
Comment (cathysiller) [inline docs/styles/demo/radios-docs.less]: remove unnecessary comment

File: docs/styles/demo/radios-docs.less [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+        // border: 1px dashed purple;
```
Comment (cathysiller) [inline docs/styles/demo/radios-docs.less]: remove unnecessary comment, unless its needed?

File: src/helix-ui/styles/components/radios.less [hunk 22]
```diff
@@ -0,0 +1,262 @@
+@import (reference) 'mixins';
+@import (reference) 'vars';
+
+// base styles
```
Comment (cathysiller) [inline src/helix-ui/styles/components/radios.less]: pls remove lines 4-10, guessing these were used for guidance of css placement but not needed anymore?

File: src/helix-ui/styles/components/radios.less [hunk 23]
```diff
@@ -0,0 +1,262 @@
...
+      }
+    }
+
+    &:checked {
+      color: @cyan-900;
+
+      ~ label > hx-radio {
+        border: 1px solid currentColor;
+        color: @cyan-900;
+        border-radius: 2rem;
```
Comment (cathysiller) [inline src/helix-ui/styles/components/radios.less]: pls move above line 127 (alphabetize)

File: src/helix-ui/styles/components/radios.less [hunk 24]
```diff
@@ -0,0 +1,262 @@
...
+        margin: 0;
+        position: relative;
+
+        &::after {
+          content: "";
```
Comment (cathysiller) [inline src/helix-ui/styles/components/radios.less]: pls alphabetize this block (lines 134-142)

File: src/helix-ui/styles/components/radios.less [hunk 25]
```diff
@@ -0,0 +1,262 @@
...
+}
+
+// LAYOUT GEOMETRY
+hx-radio-control {
+  -ms-grid-column-align: start;
+  -ms-grid-columns: auto 1fr;
+  -ms-grid-rows: auto auto;
+  display: -ms-grid;
+  display: grid;
+  grid-template-rows: auto auto;
```
Comment (cathysiller) [inline src/helix-ui/styles/components/radios.less]: pls alphabetize lines 31-33

File: src/helix-ui/styles/components/radios.less [hunk 26]
```diff
@@ -0,0 +1,262 @@
...
+    grid-area: help;
+    margin: 0;
+  }
+
+  > input[type="radio"] {
+    -ms-grid-column: 1;
+    -ms-grid-row: 1;
+    grid-area: ctrl;
+    margin: 0;
+
```
Comment (cathysiller) [inline src/helix-ui/styles/components/radios.less]: more unnecessary space

File: src/helix-ui/styles/components/radios.less [hunk 27]
```diff
@@ -0,0 +1,262 @@
...
+  > p {
+    -ms-grid-column: 2;
+    -ms-grid-row: 2;
+    grid-area: help;
+    margin: 0;
+  }
+
+  > input[type="radio"] {
+    -ms-grid-column: 1;
```
Comment (cathysiller) [inline src/helix-ui/styles/components/radios.less]: pls alphabetize lines 72-81

File: docs/components/radios/test.html [hunk 28]
```diff
@@ -11,6 +12,958 @@
...
+      <hx-radio-control>
+        <input type="radio" id="radDisabled" disabled />
+        <label for="radDisabled">
+          <hx-radio></hx-radio>
+          [disabled]
+        </label>
+      </hx-radio-control>
+    </div>
+
+    <h2>Radio Button Groups</h2>
```
[the remark this answers is not in the document]
Comment (100stacks) [reply docs/components/radios/test.html]: done

File: docs/components/radios/test.html (hunk 18 above)
Comment (100stacks) [reply docs/components/radios/test.html]: done

File: docs/components/radios/test.html (hunk 19 above)
Comment (100stacks) [reply docs/components/radios/test.html]: done

**Comment by araiford** [comment]
Hey @100stacks ,

The radio control is suffering from the vertical offset issue we experienced with checkboxes.  

image

I believe the (temporary) solution that was implemented was to set a top margin of 0.25rem on the radio to get it more lined up with the label.

Once the radio and the label are more aligned, I'll give this the LGTM.

Drew

File: docs/styles/demo/radios-docs.less (hunk 20 above)
Comment (100stacks) [reply docs/styles/demo/radios-docs.less]: comments removed

File: docs/styles/demo/radios-docs.less (hunk 21 above)
Comment (100stacks) [reply docs/styles/demo/radios-docs.less]: comment removed

File: src/helix-ui/styles/components/radios.less (hunk 22 above)
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: comments removed

File: src/helix-ui/styles/components/radios.less (hunk 23 above)
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: reordered property

File: src/helix-ui/styles/components/radios.less (hunk 24 above)
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: block alphabetized

File: src/helix-ui/styles/components/radios.less (hunk 26 above)
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: extra space removed

File: src/helix-ui/styles/components/radios.less (hunk 27 above)
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: block alphabetized

File: src/helix-ui/styles/components/radios.less (hunk 25 above)
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: Kept `grid-template-areas` after `grid-template-columns` and `grid-template-rows` based on other usage patterns.

**Comment by 100stacks** [comment]
Great catch @araiford! 

I updated the stying to compensate for the vertical styling discussed in  https://github.com/rackerlabs/helix-ui/pull/455#issuecomment-477637711.

```css
> hx-radio {
      -ms-grid-column: 1;
      -ms-grid-row: 1;
      grid-area: facade;
      display: none;
      margin-top: 0.25rem;
    }
```

File: docs/components/radios/test.html [hunk 29]
```diff
@@ -11,6 +12,958 @@
     </nav>
   </section>
 
-  <p>TBD...</p>
+  <section>
+    <header>
+      <h2>Radio Button States</h2>
```
[the remark this answers is not in the document]
Comment (100stacks) [reply docs/components/radios/test.html]: Thanks @cathysiller, I'm going to add radio groups in Phase 2.

File: docs/_data/nav.json5 [hunk 30]
```diff
@@ -83,7 +83,7 @@
             { label: '<hx-pill>', path: 'hx-pill' },
             { label: '<hx-popover>', path: 'hx-popover' },
             { label: '<hx-progress>', path: 'hx-progress' },
-            { label: '<hx-radio>', path: 'hx-radio' },
+            { label: '<hx-radio-control>', path: 'hx-radio-control' },
```
[the remark this answers is not in the document]
Comment (100stacks) [reply docs/_data/nav.json5]: Let's discuss for Phase 2, please.

**Comment by araiford** [comment]
Thanks @100stacks

File: docs/components/radios/index.html [hunk 31]
```diff
@@ -9,6 +9,60 @@
...
+              </label>
+            </hx-checkbox-control>
+            <hx-checkbox-control>
+              <input id="radIsRequired" type="checkbox" v-model="isRequired" />
+              <label for="radIsRequired">
+                <hx-checkbox></hx-checkbox>
+                Required
+              </label>
+              <p>
+                <hx-icon type="info-circle"></hx-icon> Chrome and Safari do not render <code>required</code> styling for 1 radio.
```
[the remark this answers is not in the document]
Comment (100stacks) [reply docs/components/radios/index.html]: completed

File: docs/components/radios/index.html [hunk 32]
```diff
@@ -9,6 +9,60 @@
...
+                Disabled
+              </label>
+            </hx-checkbox-control>
+            <hx-checkbox-control>
+              <input id="radIsRequired" type="checkbox" v-model="isRequired" />
+              <label for="radIsRequired">
+                <hx-checkbox></hx-checkbox>
+                Required
+              </label>
+              <p>
```
[the remark this answers is not in the document]
Comment (100stacks) [reply docs/components/radios/index.html]: completed

File: docs/_data/nav.json5 (hunk 30 above)
[the remark this answers is not in the document]
Comment (100stacks) [reply docs/_data/nav.json5]: completed

File: src/helix-ui/styles/components/radios.less [hunk 33]
```diff
@@ -0,0 +1,264 @@
+@import (reference) 'mixins';
+@import (reference) 'vars';
+
+// base styles
```
[the remark this answers is not in the document]
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: completed

File: src/helix-ui/styles/components/radios.less [hunk 34]
```diff
@@ -0,0 +1,264 @@
...
+    -ms-grid-row: 2;
+    grid-area: help;
+    margin: 0;
+  }
+
+  > input[type="radio"] {
+    -ms-grid-column: 1;
+    -ms-grid-row: 1;
+    border-radius: 1rem;
+    display: inline-block;
```
[the remark this answers is not in the document]
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: removed

File: src/helix-ui/styles/components/radios.less [hunk 35]
```diff
@@ -0,0 +1,264 @@
...
+        color: @cyan-500;
+      }
+    }
+
+    &:checked {
+      color: @cyan-900;
+
+      ~ label > hx-radio {
+        border: 1px solid currentColor;
+        border-radius: 2rem;
```
[the remark this answers is not in the document]
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: ok, removed

File: src/helix-ui/styles/components/radios.less [hunk 36]
```diff
@@ -0,0 +1,264 @@
...
+          position: absolute;
+          top: 50%;
+          transform: translate(-50%, -50%);
+          width: 0.5rem;
+        }
+      }
+    }
+
+    &:valid {
+      ~ label > hx-radio:hover {
```
[the remark this answers is not in the document]
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: selectors combined

File: src/helix-ui/styles/components/radios.less [hunk 37]
```diff
@@ -0,0 +1,264 @@
...
+        &::after {
+          content: "";
+          background-color: currentColor;
+          border-radius: 1rem;
+          left: 50%;
+          height: 0.5rem;
+          position: absolute;
+          top: 50%;
+          transform: translate(-50%, -50%);
+          width: 0.5rem;
```
[the remark this answers is not in the document]
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: ok, sounds good

File: src/helix-ui/styles/components/selectors.less [hunk 38]
```diff
@@ -104,6 +104,7 @@
     }//input
   }//label
 
+  //TODO: keep this styling?
```
[the remark this answers is not in the document]
Comment (100stacks) [reply src/helix-ui/styles/components/selectors.less]: Sounds good.  I wanted to double-check this.

File: src/helix-ui/styles/components/radios.less [hunk 39]
```diff
@@ -0,0 +1,264 @@
...
+// invalid
+// disabled
+// modern supports
+
+// BASE STYLES
+hx-radio {
+  background-color: @gray-0;
+  border: 1px solid @gray-500;
+  border-radius: 1rem;
+  color: @gray-500;
```
[the remark this answers is not in the document]
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: `color`, `background-color`, and `border` properties moved.

File: src/helix-ui/styles/components/radios.less [hunk 40]
```diff
@@ -0,0 +1,255 @@
...
+// LAYOUT GEOMETRY
+hx-radio-control {
+  -ms-grid-column-align: start;
+  -ms-grid-columns: auto 1fr;
+  -ms-grid-rows: auto auto;
+  display: -ms-grid;
+  display: grid;
+  grid-template-columns: auto 1fr;
+  grid-template-rows: auto auto;
+  grid-template-areas:
```
Comment (CITguy) [inline src/helix-ui/styles/components/radios.less]: alphabetize property

File: src/helix-ui/styles/components/radios.less [hunk 41]
```diff
@@ -0,0 +1,255 @@
...
+
+    // grid parent
+    -ms-grid-column: auto 1fr;
+    -ms-grid-row: auto;
+    display: -ms-grid;
+    display: grid;
+    grid-template-areas: 'facade txt';
+    grid-template-columns: auto 1fr;
+    grid-template-rows: auto;
+    grid-gap: 0.5rem;
```
Comment (CITguy) [inline src/helix-ui/styles/components/radios.less]: please alphabetize

File: src/helix-ui/styles/components/radios.less [hunk 42]
```diff
@@ -0,0 +1,255 @@
...
+      display: none;
+      margin-top: 0.25rem;
+    }
+  }
+
+  > p {
+    -ms-grid-column: 2;
+    -ms-grid-row: 2;
+    grid-area: help;
+    margin: 0;
```
Comment (CITguy) [inline src/helix-ui/styles/components/radios.less]: remove property, already done above with `> * { ... }`

File: src/helix-ui/styles/components/radios.less [hunk 43]
```diff
@@ -0,0 +1,255 @@
...
+    -ms-grid-row: 1;
+    border-radius: 1rem;
+    grid-area: ctrl;
+    height: 1rem;
+    margin: 0.25rem 0.5rem 0 0;
+    width: 1rem;
+
+    &::-ms-check {
+      box-sizing: border-box;
+      display: block;
```
Comment (CITguy) [inline src/helix-ui/styles/components/radios.less]: (not a deal breaker) Do we need this `display: block;` definition?

Maybe set a background color with `!important` and remove this line to see if it fills as expected.

File: src/helix-ui/styles/components/radios.less [hunk 44]
```diff
@@ -0,0 +1,255 @@
...
+    color: @gray-500;
+
+    &::-ms-check {
+      background-color: @gray-0;
+      border: 1px solid currentColor;
+      color: inherit;
+    }
+
+    &:focus {
+      box-shadow: @focus-glow;
```
Comment (CITguy) [inline src/helix-ui/styles/components/radios.less]: Double check in Edge that this shadow isn't showing up as a "dot" to the top left of the facade.

File: src/helix-ui/styles/components/radios.less [hunk 45]
```diff
@@ -0,0 +1,255 @@
...
+      ~ label > hx-radio {
+        border: 1px solid currentColor;
+        color: @cyan-900;
+        position: relative;
+
+        &::after {
+          content: "";
+          background-color: currentColor;
+          border-radius: 1rem;
+          left: 50%;
```
Comment (CITguy) [inline src/helix-ui/styles/components/radios.less]: Please alphabetize all properties (`left` after `height`, `content` after `border-radius`)

File: src/helix-ui/styles/components/radios.less (hunk 37 above)
[the remark this answers is not in the document]
Comment (CITguy) [reply src/helix-ui/styles/components/radios.less]: I'll leave the decision to move these (all props except `content`) up to you.

File: src/helix-ui/styles/components/radios.less [hunk 46]
```diff
@@ -0,0 +1,255 @@
...
+
+  ~ label > hx-radio {
+    border: 2px solid currentColor;
+    color: @red-900;
+  }
+
+  &:focus {
+    box-shadow: @focus-glow-invalid;
+
+    &::-ms-check {
```
Comment (CITguy) [inline src/helix-ui/styles/components/radios.less]: Is this block necessary?

File: src/helix-ui/styles/components/radios.less [hunk 47]
```diff
@@ -0,0 +1,255 @@
...
+
+    ~ label > hx-radio {
+      box-shadow: @focus-glow-invalid;
+    }
+  }
+
+  // native/ie11 styles
+  &:hover::-ms-check {
+    background-color: @red-100;
+    border: 2px solid currentColor !important;
```
Comment (CITguy) [inline src/helix-ui/styles/components/radios.less]: Please add comments describing _why_ we're using `!important` on line 180 and 181.

File: src/helix-ui/styles/components/radios.less [hunk 48]
```diff
@@ -0,0 +1,255 @@
...
+}
+
+// DISABLED STATE
+hx-radio-control > input[type="radio"]:disabled {
+  color: @gray-500;
+  cursor: not-allowed;
+
+  // ie11 styles
+  &::-ms-check {
+    background-color: @gray-100 !important;
```
Comment (CITguy) [inline src/helix-ui/styles/components/radios.less]: See note on `!important`

File: src/helix-ui/styles/components/radios.less [hunk 49]
```diff
@@ -0,0 +1,255 @@
...
+    border: 1px solid currentColor;
+    color: @gray-500;
+  }
+
+  &:focus {
+    &::-ms-check {
+      box-shadow: @focus-glow-disabled;
+    }
```
Comment (CITguy) [inline src/helix-ui/styles/components/radios.less]: Double check focus glow on test page for the facade.

cc: @CITguy

File: src/helix-ui/styles/components/radios.less [hunk 50]
```diff
@@ -0,0 +1,264 @@
...
+        color: @gray-500;
+      }
+    }
+  }
+}
+
+@supports (--modern: true) {
+  hx-radio-control {
+    > input[type="radio"] {
+      #Mix.hideInput();
```
[the remark this answers is not in the document]
Comment (CITguy) [reply src/helix-ui/styles/components/radios.less]: Check back on this once we've double checked Edge "dot" potential issue noted in comments above.

File: src/helix-ui/styles/components/radios.less (hunk 37 above)
[the remark this answers is not in the document]
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: Ok, I'll keep them props here.

File: src/helix-ui/elements/HXRadioSet.js [hunk 51]
```diff
@@ -0,0 +1,52 @@
+import { HXElement } from  './HXElement';
```
[the remark this answers is not in the document]
Comment (100stacks) [reply src/helix-ui/elements/HXRadioSet.js]: File removed

File: src/helix-ui/styles/components/radios.less (hunk 42 above)
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: Property removed.

File: src/helix-ui/styles/components/radios.less (hunk 40 above)
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: Property alphabetized.

File: src/helix-ui/styles/components/radios.less (hunk 41 above)
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: Properties alphabetized.

File: src/helix-ui/styles/components/radios.less (hunk 43 above)
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: Removed `display:block`.  It didn't appear to make a difference.

File: src/helix-ui/styles/components/radios.less (hunk 44 above)
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: I did not notice a `dot`.

File: src/helix-ui/styles/components/radios.less (hunk 45 above)
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: Ok, properties alphabetized.

File: src/helix-ui/styles/components/radios.less (hunk 46 above)
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: Looks like this block is not needed.  Removed.

File: src/helix-ui/styles/components/radios.less (hunk 47 above)
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: I removed the `!important`s from this file.

File: src/helix-ui/styles/components/radios.less (hunk 48 above)
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: I removed the `!important`s from this file.

File: src/helix-ui/styles/components/radios.less (hunk 50 above)
[the remark this answers is not in the document]
Comment (100stacks) [reply src/helix-ui/styles/components/radios.less]: I do not see a `dot` in Edge.

File: docs/styles/demo/radios-docs.less [hunk 52]
```diff
@@ -0,0 +1,61 @@
...
+          &:extend(hx-radio-control > input[type="radio"]:disabled ~ label > hx-radio:hover all);
+        }
+      }
+    }
+  }
+}
+
+@supports (--modern: true) {
+  .radios-test {
+    hx-radio-control {
```
[the remark this answers is not in the document]
Comment (100stacks) [reply docs/styles/demo/radios-docs.less]: Ok, consolidated selectors.

File: docs/styles/demo/radios-docs.less [hunk 53]
```diff
@@ -0,0 +1,61 @@
...
+    > input[type="radio"] {
+      &.mock-focus {
+        &:extend(hx-radio-control > input[type="radio"]:focus all);
+      }
+
+      &.mock-hover {
+        &:extend(hx-radio-control > input[type="radio"]:hover all);
+
+        ~ label > hx-radio {
+          &:extend(hx-radio-control > input[type="radio"]:valid ~ label > hx-radio:hover all);
```
[the remark this answers is not in the document]
Comment (100stacks) [reply docs/styles/demo/radios-docs.less]: It appears we need `:valid` for CSS Specificity to render the `.mock-hover` state.

File: docs/components/radios/test.html [hunk 54]
```diff
@@ -11,6 +12,780 @@
...
+      <thead>
+        <tr>
+          <th>:valid</th>
+          <th>:invalid</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr>
+          <td>
+            <hx-radio-control>
```
[the remark this answers is not in the document]
Comment (100stacks) [reply docs/components/radios/test.html]: Removed contents.

File: docs/components/radios/test.html [hunk 55]
```diff
@@ -11,6 +12,780 @@
...
```
[the remark this answers is not in the document]
Comment (100stacks) [reply docs/components/radios/test.html]: Content removed.

**Pull request merged**